spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindFieldSelfAssignment.java

```diff
@@ -161,7 +167,28 @@ public void sawOpcode(int seen) {
         default:
             break;
         }
+        if (seen == Const.GETFIELD) {
+            XField f = getXFieldOperand();
+            OpcodeStack.Item top = stack.getStackItem(0);
+            XField topF = top.getXField();
+            if (f == null || topF == null) {
+                return;
+            }
 
+            if ("this$0".equals(topF.getName())) {
+                parentInstanceLoadFromRegister = top.getFieldLoadedFromRegister();
+            }
+        }
     }
 
+    @Override
+    public void afterOpcode(int seen) {
+        super.afterOpcode(seen);
+
+        if (seen == Const.GETFIELD && parentInstanceLoadFromRegister > -1) {
```

Comment:
Could you replace this magic number `-1` with a `static final int` field?
Better to tell our intention to who is reading our code.